external/anomalib/anomaly_classification/task.py

```diff
@@ -0,0 +1,247 @@
+
+            try:
+                model.load_state_dict(model_data["model"])
+                logger.info("Loaded model weights from Task Environment")
+
+            except BaseException as exception:
+                raise ValueError("Could not load the saved model. The model file structure is invalid.") from exception
+
+        return model
+
+    def train(
+        self,
+        dataset: DatasetEntity,
+        output_model: ModelEntity,
+        train_parameters: TrainParameters,
+    ):
+        """
+        Train the anomaly task
+        """
+        config = self.get_config()
+        datamodule = OTEAnomalyDataModule(config=config, dataset=dataset)
+
+        # Callbacks.
+        progress = ProgressCallback(parameters=train_parameters)
+        model_monitor = ModelMonitorCallback(output_model, self.save_model)
+        callbacks = [progress, model_monitor]
+
+        self.trainer = Trainer(**config.trainer, logger=False, callbacks=callbacks)
+        self.trainer.fit(model=self.model, datamodule=datamodule)
+
+    def save_model(self, output_model: ModelEntity):
+        """
+        Save the model after training is completed.
+        """
+        config = self.get_config()
+        labels = {label.name: label.color.rgb_tuple for label in self.labels}
+        model_info = {"model": self.model.state_dict(), "config": config, "labels": labels, "VERSION": 1}
+        buffer = io.BytesIO()
+        torch.save(model_info, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        self.task_environment.model = output_model
+
+    def cancel_training(self):
+        """
+        Cancel the training `after_batch_end`. This terminates the training; however validation is
+        still performed.
+        """
+        logger.info("Cancel training requested.")
+        self.trainer.should_stop = True
+
+        # The runner periodically checks `.stop_training` file to ensure if cancellation is requested.
+        cancel_training_file_path = os.path.join(self.config.project.path, ".stop_training")
+        with open(file=cancel_training_file_path, mode="a", encoding="utf-8"):
+            pass
+
+    def infer(self, dataset: DatasetEntity, inference_parameters: InferenceParameters) -> DatasetEntity:
+        """
+        Perform inference on a dataset.
+        """
+        config = self.get_config()
+        datamodule = OTEAnomalyDataModule(config=config, dataset=dataset)
+
+        # Callbacks.
+        progress = ProgressCallback(parameters=inference_parameters)
+        inference = InferenceCallback(dataset, self.labels)
+        callbacks = [progress, inference]
+
+        self.trainer = Trainer(**config.trainer, logger=False, callbacks=callbacks)
+        self.trainer.predict(model=self.model, datamodule=datamodule)
+        return dataset
+
+    def evaluate(self, output_resultset: ResultSetEntity, _evaluation_metric: Optional[str] = None):
```

Comment:
Is it safe to change a name of parameter? What if the method will be call using kw args?